### PR TITLE
Feature/delete an item clean

### PR DESCRIPTION
**Pull request by TDManning**
This includes a refactor or the merchant delete test and adds code for creating an item and its associated resources.

### CODE REVIEW

**Comment by jimmacur** [comment]
The additions to both the item and merchant deletion functionality look great. The expanded test coverage also adds confidence that this logic is functioning as expected. Nice work on handling edge cases, such as the appropriate status codes and ensuring all dependent records are accounted for. And, thanks for cleaning up the files as you go along.

File: spec/requests/api/v1/merchants_requests_spec.rb [hunk 1]
```diff
@@ -122,25 +122,28 @@
...
-      item2 = merchant.items.create!(name: "Item 2", unit_price: 30)
-      customer = Customer.create!(first_name: "John", last_name: "Doe")
-      invoice1 = Invoice.create!(merchant: merchant, customer: customer, status: "pending")
-      invoice2 = Invoice.create!(merchant: merchant, customer: customer, status: "completed")
-
-      expect(Merchant.all).to include(merchant)
-      expect(merchant.items).to include(item1, item2)
-      expect(merchant.invoices).to include(invoice1, invoice2)
+    it 'can delete a merchant and associated records, returns 204 no content' do
+      item1 = @merchant_1.items.create!(name: "Item 1", unit_price: 20)
```
Comment (CCirbo) [inline spec/requests/api/v1/merchants_requests_spec.rb:126]: I love that you refactored your tests to use the new before each block. So dynamic now!

File: app/controllers/api/v1/items_controller.rb [hunk 2]
```diff
@@ -19,12 +19,19 @@ def update
     render json: ItemSerializer.new(updated_item)
    end
 
-
   def create
     new_item = Item.create(item_params)
     render json: ItemSerializer.new(new_item), status: 201
   end
 
+  def destroy
```
Comment (CCirbo) [inline app/controllers/api/v1/items_controller.rb:27]: Looks complete!

File: config/routes.rb [hunk 3]
```diff
@@ -12,12 +12,13 @@
...
   patch "/api/v1/merchants/:id", to: "api/v1/merchants#update" 
   post "/api/v1/merchants", to: "api/v1/merchants#create"
-
   delete "/api/v1/merchants/:id", to: "api/v1/merchants#destroy"
+
   get "/api/v1/items", to: "api/v1/items#index"
   get "/api/v1/items/:id", to: "api/v1/items#show"
   patch "/api/v1/items/:id", to: "api/v1/items#update" 
   post "/api/v1/items", to: "api/v1/items#create"
+  delete "/api/v1/items/:id", to: "api/v1/items#destroy"
```
Comment (CCirbo) [inline config/routes.rb:22]: Good, clean route!

**Comment by CCirbo** [comment]
Thanks for the updates! Everything looks solid, especially with the route addition and the destroy method in the ItemsController. Just a quick question—are we handling the case where the item doesn’t exist in the controller? Might be good to ensure we’re returning the appropriate error response. The test updates seem comprehensive, but maybe we could add a test for that edge case as well.

Great job overall!
